modules/ocf_kubernetes/templates/master/loadbalancer/haproxy.cfg.erb

```diff
@@ -0,0 +1,60 @@
+global
+  daemon
+  log localhost local0
+  # TODO: send to syslog
+  maxconn 4000
+  pidfile /var/run/haproxy.pid
+  user haproxy
+  group sys
+  chroot /var/lib/haproxy
+
+defaults
+  timeout connect 10s
+  timeout client 30s
+  timeout server 30s
+  option httplog
+  log global
+  option redispatch
+  retries 3
+
+frontend lb-http
+  bind 0.0.0.0:80
+  mode http
+
+  # https redirect
+  redirect scheme https code 301 if !{ ssl_fc }
+
+frontend lb-https
+  bind 0.0.0.0:443 ssl crt /etc/ssl/private/<%= @fqdn %>.pem
+  mode http
+
+  <%- @kubernetes_aliases.each do |service_name, aliases| -%>
+  redirect prefix https://<%= service_name %> code 301 if { hdr(host) -i <%= aliases.join(' ') %> }
+  use_backend kubernetes if { hdr(host) -i <%= service_name %> }
+  <%- end -%>
+
+  # send *.dev-kubernetes.ocf.berkeley.edu into the cluster
+  use_backend kubernetes if { hdr_end(host) -i dev-kubernetes.ocf.berkeley.edu }
+
+
+# Some services don't need SSL termination
+# Either because it has already been done (ocfweb, irc)
+# Or because it doesn't need it (snmp_exporter)
+# As all such traffic should be internal, it is placed on a firewall'd port
```

Comment:
I'd make this clear that it's an externally firewalled port (using the firewall the university gave us), not a local one (with iptables), since we do use both.

Also, something to note here, but the thing proxying to this location must have done the canonicalization step already (for instance rewriting `irc.ocf.io` to `irc.ocf.berkeley.edu`), so I'm not sure if that's worth mentioning here or if it's worth including anything to redirect those cases that slip through proxies.

Reply:
Fixed.
There are just 4 of them, and all of them do the canonicalization step step already, so I don't think it's necessary to add that code.